### PR TITLE
Refactor ingestor code

### PR DESCRIPTION
**Pull request by dejanb**
# Description of the PR

While looking into adding a compression, I noticed that ingestion code has a lot of repetitions in different collectors and services, so thought to try to consolidate things to make it easier to modify as we move forward.

This PR:
* Introduces new `ingestor` package that extracts all common logic into one place
* Introduces `ingestor.Ingest` function that can be used by different collectors to synchronously ingest `Document` using GraphQL
* Changes all existing collectors to use it
* Refactor `guacingest` tool to be able to use both sync and async ingestion (as async ingestion send data over nats multiple times which could be an issue with large files)

# PR Checklist

- [x] All commits have [a Developer Certificate of Origin (DCO)](https://wiki.linuxfoundation.org/dco) -- they are generated using `-s` flag to `git commit`.
- [ ] All new changes are covered by tests
- [ ] If GraphQL schema is changed, `make generate` has been run
- [ ] If `collectsub` protobuf has been changed, `make proto` has been run
- [x] All CI checks are passing (tests and formatting)
- [ ] All dependent PRs have already been merged

### CODE REVIEW

**Review by pxp928 — COMMENTED** [review]
quick nits. Will look into detail. Thanks @dejanb!

File: cmd/guacingest/cmd/ingest.go [hunk 1]
```diff
@@ -80,69 +79,65 @@ func ingest(cmd *cobra.Command, args []string) {
...
+		}
 	}
 
 	// Assuming that publisher and consumer are different processes.
 	var wg sync.WaitGroup
 	wg.Add(1)
 	go func() {
 		defer wg.Done()
-		if err := processorFunc(); err != nil {
+		if err := process.Subscribe(ctx, emit); err != nil {
```
Comment (pxp928) [inline cmd/guacingest/cmd/ingest.go:112]: in this case, even if it's not async, you will still need NATS running for this to work?

Comment (pxp928) [reply cmd/guacingest/cmd/ingest.go:112]: meaning if `return ingestor.Ingest(ctx, d, opts.graphqlEndpoint, csubClient)` is running, why are we throwing it back into NATS here?

Comment (dejanb) [reply cmd/guacingest/cmd/ingest.go:112]: Yes, this is the ingestor the receives documents via NATS. The only thing that async flag is doing is preventing it from sending it back via pub/sub via stages. 

So, sync scenario:

`doc -> nats -> ingestor.Ingest`

async scenario:

`doc -> nats -> processor.Process -> nats -> parser.Parse -> assembler`

I'm not sure if async scenario is having much sense any more, but I didn't want to remove it without further discussion.

Comment (lumjjb) [reply cmd/guacingest/cmd/ingest.go:112]: yea i am in favor of removing async for now unless we run into the need to granularly scale the processor and parser aspects, which i doubt we would need to in the near future.

File: cmd/guacone/cmd/gcs.go [hunk 2]
```diff
@@ -93,46 +93,17 @@ var gcsCmd = &cobra.Command{
...
-			err = collectSubEmitFunc(idstrings)
-			if err != nil {
-				logger.Infof("unable to create entries in collectsub server, but continuing: %v", err)
-			}
+			err := ingestor.Ingest(ctx, d, opts.graphqlEndpoint, csubClient)
 
-			err = assemblerFunc(predicates)
 			if err != nil {
 				gotErr = true
-				return fmt.Errorf("unable to assemble graphs: %v", err)
```
Comment (lumjjb) [inline cmd/guacone/cmd/gcs.go:131]: nit :%w (we missed it in the previous code) - multiple occurrences

File: docker-compose.yml [hunk 3]
```diff
@@ -88,7 +88,7 @@ services:
   guac-ingestor:
     networks: [frontend]
     image: $GUAC_IMAGE
-    command: "/opt/guac/guacingest"
```
Comment (lumjjb) [inline docker-compose.yml:91]: can we add this to guac.yaml instead?

**Review by lumjjb — APPROVED** [review]
Some nits, else LGTM

**Comment by dejanb** [comment]
@pxp928 @lumjjb thanks for the review. I ended up removing async ingest option per previous comments. It simplify things considerebly.

File: pkg/cli/store.go [hunk 4]
```diff
@@ -71,6 +71,7 @@ func init() {
 	set.Bool("service-poll", true, "sets the collector or certifier to polling mode")
 	set.BoolP("poll", "p", false, "sets the collector or certifier to polling mode")
 	set.StringP("interval", "i", "5m", "if polling set interval, m, h, s, etc.")
+	set.Bool("async-ingest", true, "sets the ingestor in async mode, meaning document will be sent via pub/sub between ingestion stages")
```
Comment (pxp928) [inline pkg/cli/store.go:74]: this flag is no longer needed

**Review by pxp928 — APPROVED** [review]
couple of small nits else LGTM! Thanks @dejanb

**Comment by dejanb** [comment]
> couple of small nits else LGTM! Thanks @dejanb

Thanks @pxp928 . I think all of them should be addressed now

**Comment by pxp928** [comment]
@dejanb Looks like you missed a sign-off on the last commit and it will auto merge

**Comment by dejanb** [comment]
> @dejanb Looks like you missed a sign-off on the last commit and it will auto merge

@pxp928 fixed

**Review by pxp928 — APPROVED** [review]

**Pull request merged**